### DISCUSSION TITLE
Updating an Entity with Hibernate Reactive and Panache

### DISCUSSION THREAD

**Discussion by jack-farley** [quarkusio/quarkus#32019] [Q&A] [answered]
I am trying to update an entity with hibernate reactive and panache. I saw in the documentation that if you have a persistent entity (i.e. obtained by fetching using findById) any modifications would be persisted when the transaction is committed. However, this doesn't seem to be working. Here is my attempt at updating a user profile object:

```
    @ReactiveTransactional
    public Uni<UserProfile> updateProfile(UserProfile updatedProfile) {
        return UserProfile.<UserProfile>findById(updatedProfile.getProfileId())
                .invoke(originalProfile -> {
                    originalProfile.firstName = updatedProfile.firstName;
                    originalProfile.lastName = updatedProfile.lastName;
                    originalProfile.phoneNumber = updatedProfile.phoneNumber;
                    originalProfile.dateOfBirth = updatedProfile.dateOfBirth;
                });
    }
```

Here is my profile definition: 

```
@Entity
@Getter
public class UserProfile extends PanacheEntityBase {

    @Id
    @Column(name = "id")
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID",
            strategy = "org.hibernate.id.UUIDGenerator")
    UUID profileId;

    String firstName;

    String lastName;

    String phoneNumber;

    LocalDate dateOfBirth;
}
```

I would appreciate any help understanding why this doesn't work. Also, is there a better way to go about making updates? It seems like manually updating each field of the profile is not the best.

[a comment by an automated account is not shown]

**Comment by DavideD** [comment]
Which version of Quarkus are you using?

**Reply by jack-farley** [reply]
2.15.3.Final

**Comment by jack-farley** [comment]
Any ideas on this?

**Comment by DavideD** [comment]
I haven't had the time to create a test case. If you could provide a project that shows the error, I will look into it

**Reply by jack-farley** [reply]
Figured out the problem while setting up an example, wrote it up in another comment. I'm not sure if the need for PanacheEntity fields to be public is mentioned in the documentation and I missed it, but if not, I think it would be helpful to add a note of this.

Edit: Looks like it is there and I just missed it.

**Comment by jack-farley** [comment, marked as answer]
I think I figured out the problem. It seems like when you have fields in the entity class that are not explicitly marked 'public', they will not get updated when you try to update them as part of a transaction. Marking all the fields of the panache entity as public fixes the problem.
